ibm/service/kubernetes/resource_ibm_container_vpc_worker_pool.go

```diff
@@ -515,6 +514,30 @@ func resourceIBMContainerVpcWorkerPoolUpdate(d *schema.ResourceData, meta interf
 		}
 	}
 
+	if d.HasChange("operating_system") {
+		clusterNameOrID := d.Get("cluster").(string)
+		workerPoolName := d.Get("worker_pool_name").(string)
+		operatingSystem := d.Get("operating_system").(string)
+		targetEnv, err := getVpcClusterTargetHeader(d)
+		if err != nil {
+			return err
+		}
+		ClusterClient, err := meta.(conns.ClientSession).VpcContainerAPI()
+		if err != nil {
+			return err
+		}
+		Env := v2.ClusterTargetHeader{ResourceGroup: targetEnv.ResourceGroup}
+
+		err = ClusterClient.WorkerPools().SetWorkerPoolOperatingSystem(v2.SetWorkerPoolOperatingSystem{
+			Cluster:         clusterNameOrID,
+			WorkerPool:      workerPoolName,
+			OperatingSystem: operatingSystem,
+		}, Env)
+		if err != nil {
+			return fmt.Errorf("[ERROR] Error updating the operating_system %s: %s", operatingSystem, err)
+		}
```

Comment:
After we trigger operating system should n't we wait for the operation to be completed and workers comes to normal state.
If we have any dependent operations may be using kubernetes provider will be failed if we don't wait for the update to completed

Reply:
This API call only sets the "desired" operating system and it happens immediately. It does not start the update. The client needs to call the worker update/replace API which will make sure to match the workerpool's desired operating system = actual operating system. The `update_all_workers` will go one by one and do that and wait for the update to finish.

Reply:
@hkantare this is similar to the behavior we have for OS update in the API/CLI/UI. A customer can update the OS property on the worker pool, but then has to roll it out to the worker. We can add some optimizaiton in Terraform for users to opt in to the workers getting updated at the same time the pool does, but that can be added in the next release after this initial release that we need in order to give customers a full 6 months to migrate before before EOS for the exisitng OSes.